src/Razor/src/Microsoft.AspNetCore.Razor.VSCode.Extension/syntaxes/aspnetcorerazor.tmLanguage.json

```diff
@@ -311,7 +311,7 @@
           "include": "#implicit-expression-continuation"
         }
       ],
-      "end": "(?=[\\s<>])"
+      "end": "(?=[\\s<>\\{\\}\\)\\]])"
```

Comment:
What does the expression look like where we would involve `)` and `]`? Because they're not represented in the `begin` phase It looks like something like `@Thing]` would be legal? Also because we don't do capturing or backtracing wouldn't this grammar report `@(Parameter}` as legal?

Reply:
So the Regex `(?=ABC)` is used to say match when the next character(s) is/are the following. So in this case we're saying, stop parsing an implicit expression if the next character is one of the following above. This typically happens when the implicit expression is wrapped in some other construct such as `@section Foo {@DateTime.Now}`. Without this change the `}` would be treated as part of the implicit expression and colored as such as well; however, in section's case the `}` corresponds to the ending of the directive and thus should be colored as directive braces are (purple by default).